src/Generation/GirLoader/Helper/ChainedRepositoryResolver.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+namespace GirLoader;
+
+/// <summary>
+/// Tries multiple repository resolvers, returning the first result found
+/// </summary>
+public class ChainedRepositoryResolver : IRepositoryResolver
+{
+    private readonly IRepositoryResolver[] _resolvers;
+
+    public ChainedRepositoryResolver(IEnumerable<IRepositoryResolver> resolvers)
+    {
+        _resolvers = resolvers.ToArray();
```

Comment:
In general I don't like storing an `IEnumerable` member due to enumerating possibly being expensive, and the rarer problems like results being able to change between enumerations. But accepting an `IEnumerable` parameter provides maximum flexibility to calling code.

But like you say this is all internal code rather than being part of an API for external use, so that flexibility is unnecessary. This could probably just be an `IRepositoryResolver[]` parameter to be honest, there isn't really a need to accept a more general interface. But the extra overhead of using something like `IReadOnlyCollection` is going to be negligible and that better fits the intended behaviour of this member without tying it to a specific type, so I will switch to that.

And yes you're right that this will cause a copy of the array. Copying an array of two elements once on application startup didn't seem like a big deal, but yes it's better to avoid the copy if it's not needed.